src/Management/src/EndpointCore/Metrics/EndpointServiceCollectionExtensions.cs

```diff
@@ -79,7 +82,47 @@ public static void AddPrometheusActuator(this IServiceCollection services, IConf
             services.AddActuatorEndpointMapping<PrometheusScraperEndpoint>();
         }
 
-        private static void AddMetricsObservers(IServiceCollection services, MetricsObserverOptions observerOptions)
+        /// <summary>
+        /// Adds the services used by the Wavefront exporter
+        /// </summary>
+        /// <param name="services">Reference to the service collection</param>
+        /// <param name="configuration">Reference to the configuration system</param>
+        /// <returns>A reference to the service collection</returns>
+        public static IServiceCollection AddWavefrontMetrics(this IServiceCollection services, IConfiguration configuration = null)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            configuration ??= services.BuildServiceProvider().GetService<IConfiguration>();
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
```

Comment:
We might be able to avoid this and not use the IConfiguration param at all if we use this pattern:
```csharp
serviceCollection.AddSingleton(typeof(CloudFoundryApplicationOptions), serviceProvider => new CloudFoundryApplicationOptions(serviceProvider.GetRequiredService<IConfiguration>()));
```

Reply:
The main issue is with AddMetricObservers which requires an metricsOptions object created out of configuration. I am not seeing any way to add an IEnumerable that can conditionally do this on the contents of configuration.

Reply:
I don't have a great answer for that one right now, so let's build the metricsOptions object for just that one case, but use the newer pattern everywhere else that we can

Reply:
Those observers probably all need to be rebuilt using to self enable/disable at runtime based on `IOptions` rather than this startup-based yes/no option